common/primitives/core/src/intent.rs

```diff
@@ -1,16 +1,23 @@
+use crate::{AccountId, Balance};
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
 use sp_core::H160;
 use sp_runtime::{traits::ConstU32, BoundedVec};
 
 pub const CALL_ETHEREUM_INPUT_LEN: u32 = 10 * 1024;
 
+pub const MAX_REMARK_LEN: u32 = u32::max_value();
+
 #[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, MaxEncodedLen, TypeInfo)]
 pub enum Intent {
     #[codec(index = 0)]
     TransferEthereum(TransferEthereum),
     #[codec(index = 1)]
     CallEthereum(CallEthereum),
+    #[codec(index = 2)]
+    SystemRemark(BoundedVec<u8, ConstU32<MAX_REMARK_LEN>>),
+    #[codec(index = 3)]
+    TransferNative(TransferNative),
```

Comment:
While this is fine for the demo, how about having a more generic `CallNative(Box<RuntimeCall>)`?